# Problem 1

In [10]:
import numpy as np

In [11]:
g1 = np.array([60,54,46,41])
g2 = np.array([40,44,53,57])
g = np.append([g1],[g2])
sum1 = g1.sum()
sum2 = g2.sum()
total = sum1+sum2
f_per = sum1/total
m_per = sum2/total
exp_f = [np.round((g1[i]+g2[i])*f_per,2)  for i in range(4)]
exp_m = [np.round((g1[i]+g2[i])*m_per,2)  for i in range(4)]
exp_g = np.append([exp_f],[exp_m])
chi_score = np.array([((exp_g[i]-g[i])**(2))/exp_g[i] for i in range(8)]).sum()
chi_score = round(chi_score,3)
print('chi score of given data is = ',chi_score)

chi score of given data is =  8.005


Conclusion:-

    The critical value of chi square with 3 degree of freedom ((r-1)*(c-1)) is 7.815. Since 8.005 > 7.815, we reject the null hypothesis and conclude that the education level depends on gender at a 5% level of significance.

# Problem 2

Using the following data, perform a oneway analysis of variance using α=.05. Write up the results in APA format. [Group1: 51, 45, 33, 45, 67] [Group2: 23, 43, 23, 43, 45] [Group3: 56, 76, 74, 87, 56]

In [12]:
def ss(g,mean):
    if type(g)==np.ndarray:
        s = np.array([(i-mean)**(2) for i in g]).sum()
        return s  
    else:
        s = (g-mean)**(2)
        return s   

def anova(arr):
    m = arr.shape[0]
    n = arr.shape[1]
    grp_means = np.array([i.mean() for i in arr])
    g_mean = grp_means.mean()
    SST = np.round(np.array([ss(arr[i],g_mean) for i in range(m)]).sum(),2)
    SSW = np.round(np.array([ss(arr[i],grp_means[i]) for i in range(m)]).sum(),2)
    SSB = np.round(np.array([n*ss(grp_means[i],g_mean) for i in range(m)]).sum(),2)
    return SST,SSW,SSB

def f_stats(SSB,SSW):
    m = arr.shape[0]
    n = arr.shape[1]
    f = (SSB/(m-1)) / (SSW/(m*(n-1))) 
    return np.round(f,2)

In [13]:
g1 = np.array([51, 45, 33, 45, 67])
g2 = np.array([23, 43, 23, 43, 45])
g3 = np.array([56, 76, 74, 87, 56])
arr = np.append([g1,g2],[g3]).reshape(3,5)
SST,SSW,SSB = anova(arr)
f = f_stats(SSB,SSW)

In [14]:
print('Total sum of squared SST = {}\nWithin sum of squared SSW = {}\nBetween sum of squared SSB = {} and\nf statistics = {}'.format(SST,SSW,SSB,f))

Total sum of squared SST = 4883.73
Within sum of squared SSW = 1860.8
Between sum of squared SSB = 3022.93 and
f statistics = 9.75


Conclusion:-

    Critical value for α=.05 is 2.80. that means anything less than critical value accept Null hypothesis(all population mean are same). However, f value is greater than critical value we reject the Null hypothesis.  

# Problem 3

Calculate F Test for given 10, 20, 30, 40, 50 and 5,10,15, 20, 25.
For 10, 20, 30, 40, 50:

In [15]:
g1 = np.array([10,20,30,40,50])
g2 = np.array([5,10,15,20,25])
g3 = np.array([10,20,30,40,50])
arr = np.append([g1,g2],[g3]).reshape(3,5)
SST,SSW,SSB = anova(arr)
f = f_stats(SSB,SSW)
print('F test for given is = ',f)

F test for given is =  2.0
